inbm/dispatcher-agent/dispatcher/common/dispatcher_state.py

```diff
@@ -94,13 +115,13 @@ def write_dispatcher_state_to_state_file(state: DispatcherState) -> None:  # pra
     try:
         # if there's already a state file, read it in so we can update it and write back
         if is_dispatcher_state_file_exists():
-            with builtins.open(DISPATCHER_STATE_FILE, 'rb') as fd:
+            with builtins.open(NEW_DISPATCHER_STATE_FILE, 'rb') as fd:
```

Comment:
is_dispatcher_state_file_exists() checks both the OLD and NEW location of the dispatcher state file. This can cause an error if the old location exists but the function opens the new location instead.

Reply:
@yengliong93 good point.. I'll have to think about this